upload_vehicle_data.py

```diff
@@ -0,0 +1,420 @@
+    """Delete a specific file on the remote machine."""
+    command = f'rm {file_path}'
+    run_ssh_command(remote_user, remote_ip, command)
+
+
+def main(config):
+    """Automate the upload of rosbags."""
+    remote_user = config['remote_user']
+    remote_ip = config['remote_ip']
+    remote_directory = config['remote_directory']
+    cloud_upload_directory = config['cloud_upload_directory']
+    clean_up = config['clean_up']
+    max_upload_attempts = config.get(
+        'upload_attempts', 3
+    )  # Default to 3 if not specified
+    # Create the remote temporary directory
+    create_remote_directory(remote_user, remote_ip, remote_directory)
+
+    try:
+        # Get the list of rosbags from the remote machine
+        rosbag_list = get_remote_rosbags_list(
+            remote_user, remote_ip, remote_directory
+        )
+
+        bandwidth_mbps = measure_bandwidth(remote_ip, remote_user)
+        if bandwidth_mbps is None:
+            print('Could not measure bandwidth. Exiting.')
+            return
+
+        total_size_gb = sum(
+            get_remote_file_size(remote_user, remote_ip, rosbag)
+            for rosbag in rosbag_list
+        )
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps
+        )
+
+        print(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        print(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+        print(f'Estimated upload time: {estimated_time:.2f} hours.')
+        confirm = input('Do you want to proceed to upload? (yes/no): ')
+
+        if confirm.lower() != 'yes':
+            print('Upload aborted.')
+            return
+
+        logging.info(
+            f'Starting upload of {len(rosbag_list)} files with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        logging.info(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+
+        successfully_uploaded_files = []
+
+        for rosbag in rosbag_list:
+            success = compress_and_transfer_rosbag(
+                remote_user,
+                remote_ip,
+                rosbag,
+                remote_directory,
+                cloud_upload_directory,
+                max_upload_attempts,
+            )
+            if not success:
+                logging.error(
+                    f'Failed to process {rosbag}. Continuing with next file.'
+                )
+            else:
+                successfully_uploaded_files.append(rosbag)
+                logging.info(f'Successfully processed {rosbag}.')
+
+        if clean_up:
+            # Only delete rosbag files that were successfully uploaded
+            for rosbag in successfully_uploaded_files:
+                delete_remote_file(remote_user, remote_ip, rosbag)
+
+    finally:
+        # Delete the remote temporary directory
+        delete_remote_directory(remote_user, remote_ip, remote_directory)

```

Comment:
This finally shouldn't delete the remote directory, as it could delete uncompressed rosbags that were not successfully uploaded. Could you please remove it?